spring-apps/autosleep-core/src/main/java/org/cloudfoundry/autosleep/ui/web/controller/AutoEnrollmentController.java

```diff
@@ -0,0 +1,143 @@
+
+    @Autowired
+    private EnrollmentValidator validator;
+
+    @Autowired
+    private CloudFoundryApiService cloudfoundryApi;
+
+    @Autowired
+    private ParameterReader<Config.ServiceInstanceParameters.Enrollment> autoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Pattern> excludeFromAutoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Duration> idleDurationReader;
+
+    @Autowired
+    private ParameterReader<EnrollmentConfig.EnrollmentParameters.EnrollmentState> stateReader;
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> getEnrolledOrganization(
+            @PathVariable String organizationId) {
+        log.debug("getEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<OrgEnrollmentConfig>(orgConfig, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> deleteOrganization(
+            @PathVariable String organizationId) {
+        log.debug("deleteEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        orgEnrollmentRepository.delete(organizationId);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> enrolOrganization(
+            @PathVariable String organizationId,
+            @RequestBody OrgEnrollmentConfigRequest orgEnrollerConfigRequest, BindingResult result)
+            throws CloudFoundryException, BindException {
+
+        log.debug("enrolOrganization - {}", organizationId);
+        if (!cloudfoundryApi.isValidOrganization(organizationId)) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        validator.validate(orgEnrollerConfigRequest, result);
+        if (result.hasErrors()) {
+            throw new BindException(result);
+        }
+
+        OrgEnrollmentConfig orgEnrollerConfig = OrgEnrollmentConfig.builder()
+                .organizationGuid(organizationId)
+                .idleDuration(orgEnrollerConfigRequest.getIdleDuration() == null ? null
+                        : idleDurationReader
+                                .readParameter(orgEnrollerConfigRequest.getIdleDuration(), false))
+                .excludeSpacesFromAutoEnrollment(
+                        orgEnrollerConfigRequest.getExcludeSpacesFromAutoEnrollment() == null ? null
+                                : excludeFromAutoEnrollmentReader
```

Comment:
The REST API implementation currently only validates the input that the user provides. If the input is valid, it persists the data into the table or else returns a BAD REQUEST response. It does not modify the data by including the defaults so that the end user can GET what he had initially PUT.

`>> we would like new environment variables in the manifest.yml to specify defaults when the REST API did not receive them`
The reading of the default values from the manifest will be built into the poller that will be responsible for creating the service instances based on the enrollment data. I will create a tracker story to track this. Need not necessarily be part of the REST API implementation. 

`JSON to POJO conversion will provide such built-in validation`
The ParameterReaderFactory can be used to validate and return the defaults for the autosleep parameters. Maybe it would be a good idea to pull this out of the servicebroker package into a more reusable area. The reason why we put this validator and at the same time, the OrgEnrollmentConfigRequest class was to accept the input that the user had provided via the REST API, validate the input in its entirety and then return back the response with all the errors. The current approach with the service broker returns back the response on identifying the first error in the data even if there could be more. Hence "@ valid" does not work for us if we want to accept the request object and validate it completely before returning.